src/apps/loan-list/materials/stackable-material/stackable-material.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useCallback, FC, MouseEvent, useState } from "react";
+import { useText } from "../../../../core/utils/text";
+import {
+  FetchMaterial,
+  StackableMaterialProps,
+  MaterialProps
+} from "../utils/material-fetch-hoc";
+import { useModalButtonHandler } from "../../../../core/utils/modal";
+import DueDateLoansModal from "../../modal/due-date-loans-modal";
+import MaterialStatus from "./material-status";
+import MaterialOverdueLink from "./material-overdue-link";
+import AdditionalMaterialsButton from "./additional-materials-button";
+import MaterialInfo from "./material-info";
+
+const StackableMaterial: FC<StackableMaterialProps & MaterialProps> = ({
+  amountOfMaterialsWithDueDate,
+  material,
+  openModal,
+  selectMaterial,
+  loanMetaData,
+  dueDateLabel,
+  stack
+}) => {
+  const t = useText();
+  const { open } = useModalButtonHandler();
+  const [additionalMaterials] = useState(
+    amountOfMaterialsWithDueDate ? amountOfMaterialsWithDueDate - 1 : 0
+  );
+  const [showModal, setShowModal] = useState(false);
+  const { dueDate, id } = loanMetaData;
+
+  function stopPropagationFunction(e: Event | MouseEvent) {
+    e.stopPropagation();
+  }
+  useEffect(() => {
+    document
+      .querySelector(".list-reservation a")
+      ?.addEventListener("click", stopPropagationFunction, true);
+
+    return () => {
+      document
+        .querySelector(".list-reservation a")
+        ?.removeEventListener("click", stopPropagationFunction, true);
+    };
+  }, []);
+
+  const openDueDateModal = useCallback(() => {
+    if (stack && dueDate) {
+      setShowModal(true);
+      open(dueDate);
+    }
+  }, [stack, open, dueDate]);
+
+  useEffect(() => {
+    if (openModal) {
+      openDueDateModal();
+    }
+  }, [openDueDateModal, openModal]);
+
+  const selectListMaterial = useCallback(
+    (e: MouseEvent) => {
+      stopPropagationFunction(e);
+      if (selectMaterial) {
+        selectMaterial({
```

Comment:
Just a suggestion:
I usually name callbacks: "*Handler" to emphasize the purpose of the prop.
So in this case it would be: "selectMaterialHandler"